Client/reasn-client/packages/common/models/EventDto.ts

```diff
@@ -0,0 +1,80 @@
+        this.Name = name
+        this.AddressId = addressId
+        this.Description = description
+        this.OrganizerId = organizerId
+        this.StartAt = startAt
+        this.EndAt = endAt
+        this.CreatedAt = createdAt
+        this.UpdatedAt = updatedAt
+        this.Slug = slug
+        this.StatusId = statusId
+        this.Tags = tags
+    }
+
+    static fromJson(json: string): EventDto | null {
+        if (!json) {
+            return null
+        }
+
+        return EventDto.fromObject(JSON.parse(json))
+    }
+
+    static fromObject(obj: object): EventDto | null {
+        if (!obj) {
+            return null
+        }
+
+        if ('Name' in obj === false || typeof obj.Name !== 'string') throw new ModelMappingError('EventDto','Name is required')
+        if ('AddressId' in obj === false || typeof obj.AddressId !== 'number') throw new ModelMappingError('EventDto','AddressId is required')
+        if ('Description' in obj === false || typeof obj.Description !== 'string') throw new ModelMappingError('EventDto','Description is required')
+        if ('OrganizerId' in obj === false || typeof obj.OrganizerId !== 'number') throw new ModelMappingError('EventDto','OrganizerId is required')
+        if (!('StartAt' in obj) || (typeof obj.StartAt !== 'string' && !(obj.StartAt instanceof Date))) throw new ModelMappingError('CommentDto','StartAt is required')
+        if (typeof obj.StartAt === 'string' && isNaN(Date.parse(obj.StartAt))) throw new ModelMappingError('CommentDto','StartAt is invalid')
+        if (!('EndAt' in obj) || (typeof obj.EndAt !== 'string' && !(obj.EndAt instanceof Date))) throw new ModelMappingError('CommentDto','EndAt is required')
+        if (typeof obj.EndAt === 'string' && isNaN(Date.parse(obj.EndAt))) throw new ModelMappingError('CommentDto','EndAt is invalid')
+        if (!('CreatedAt' in obj) || (typeof obj.CreatedAt !== 'string' && !(obj.CreatedAt instanceof Date))) throw new ModelMappingError('CommentDto','CreatedAt is required')
+        if (typeof obj.CreatedAt === 'string' && isNaN(Date.parse(obj.CreatedAt))) throw new ModelMappingError('CommentDto','CreatedAt is invalid')
+        if (!('UpdatedAt' in obj) || (typeof obj.UpdatedAt !== 'string' && !(obj.UpdatedAt instanceof Date))) throw new ModelMappingError('CommentDto','UpdatedAt is required')
+        if (typeof obj.UpdatedAt === 'string' && isNaN(Date.parse(obj.UpdatedAt))) throw new ModelMappingError('CommentDto','UpdatedAt is invalid')
+        if ('Slug' in obj === false || (typeof obj.Slug !== 'string')) throw new ModelMappingError('EventDto','Slug is required')
+        if ('StatusId' in obj === false || typeof obj.StatusId !== 'number') throw new ModelMappingError('EventDto','StatusId is required')
+        if ('Tags' in obj === false || obj.Tags === null || !Array.isArray(obj.Tags)) throw new ModelMappingError('EventDto','Tags is required')
```

Comment:
The way to validate responses from API and creating the appropriate DTOs from them seems very complicated especially for DTOs with a large number of properties such as this one. What's more, the DTO by the definition should not have any methods so parsers or validators should be implemented as independent functions. I would consider making universal validator to somehow simplify this logic or use third party (lightweight) library to do so.